### PR TITLE
Add Testing doc

### PR DESCRIPTION
**Pull request by isidorn**
This PR adds a skeleton for the Testing doc, so we have a location to make progress on this.

@connor4312 @gregvanl where do you think should this document live? Currently I put it under Typescript, but all of this also holds true for Javascript. We do not have a big Javascript section, only nodejs? So alternatively it could be under nodejs, and the Typescript one would only link to that. What do you think?

@marcusfelling said he can help with the Playwright content

@connor4312 I added sections for the popular testing frameworks, if you think something else should also be covered let me know.

fyi @danielrosenwasser

### CODE REVIEW

**Comment by connor4312** [comment]
It'd be nice if we could link this doc under both sections, TypeScript as well as Node.js/JavaScript.

Testing frameworks looks like a good start 🙂

**Comment by gregvanl** [comment]
The Node.js/JavaScript topics get about 3x the page views of the TypeScript TOC topics so for discovery it might be better to put the page there. We can have an entry in the TypeScript TOC which jumps to the topic under Node.js/JavaScript.
Also feels right if it is a more general JS testing topic, no matter what someone uses as the initial source code.

**Comment by smcenlly** [comment]
@isidorn - our [Wallaby.js extension](https://marketplace.visualstudio.com/items?itemName=WallabyJs.wallaby-vscode) was [one of the first VS Code extensions](https://twitter.com/ArtemGovorov/status/667136732660297728) in the marketplace. While we provide a commercial product, we also provide [free licenses for open source projects](https://wallabyjs.com/oss/) and think it could be really beneficial to mention us for JavaScript/TypeScript testing. Thinking a little more about it, would be good to list a number of popular extensions directly on the page to help developers get started.

If you agree on adding test runner content, we'd be happy to write a section about Wallaby for you.

**Comment by isidorn** [comment]
@gregvanl I love that idea. Moving the doc there then.

@smcenlly thanks for reaching out. I like your idea that we should list all the test extensions. Maybe we can have a section at the end that lists all the others.
As for the Wallaby.js extension I really like your extension and I know you were one of the first extensions we had :) For this doc I would prefer to promote extensions that use our new [Testing API](https://code.visualstudio.com/api/extension-guides/testing). And as far as I understand Wallaby.js is not using this, right? Are there plans to adopt to the new Testing API?

**Comment by smcenlly** [comment]
@isidorn - we started using the testing API very shortly after it was released and were an early adopter. I also made some contributions to the Testing API shortly after it was released (https://github.com/microsoft/vscode/pull/132097, https://github.com/microsoft/vscode/pull/132099).

**Comment by isidorn** [comment]
@smcenlly I was not aware of this, thanks for clarifying! 
Then a section in the doc would make sense. I would like the section to have 1 image, to be concise, have a neutral tone, and for more details can point to your docs. 
I would do a final review and update to the docs.

If that works for you then a contribution would be very welcome :)

**Review by 404Nikhil — APPROVED** [review]

**Comment by smcenlly** [comment]
I couldn't work out how to make an edit to this pull request but have proposed an adjustment to `docs/nodejs/nodejs-testing.md` below so that this page can be merged to docs in a completed state.

---

# Testing Node.js

Visual Studio Code supports testing Node.js and TypeScript through it's testing framework. You will need to have the right  [testing extension](https://marketplace.visualstudio.com/search?target=VSCode&category=Testing) based on the testing framework that your project uses.

## Mocha

The [Mocha Test Explorer](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-mocha-test-adapter) allows you to run your [Mocha](https://mochajs.org) tests in the Sidebar of Visual Studio Code using the Test Explorer UI. 

Mocha Screenshot

## Playwright

[Playwright Test for VSCode](https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright) allows you to run [Playwright](https://playwright.dev) tests in Visual Studio Code.

Playwright Screenshot

## Jest

The [Jest VS Code Extension](https://marketplace.visualstudio.com/items?itemName=Orta.vscode-jest) allows you to run your [Jest](https://jestjs.io) tests in Visual Studio Code.

Jest Screenshot

## Vitest

The [Vitest VS Code Extension](https://marketplace.visualstudio.com/items?itemName=ZixuanChen.vitest-explorer) allows you to run your [Vitest](https://vitest.dev) tests in Visual Studio Code.

Vitest Screenshot

## Other extensions

There are other extensions that support testing in Visual Studio Code. You can find them in the [Visual Studio Code Marketplace](https://marketplace.visualstudio.com/search?target=VSCode&category=Testing). Some popular ones are listed below.

### Wallaby.js

[Wallaby.js](https://wallabyjs.com) works with most JavaScript testing frameworks and runs your tests immediately as you type, highlighting results right next to your code. It is a paid product for commercial use, but is free for open source projects.

Wallaby.js Screenshot

**Comment by isidorn** [comment]
@smcenlly thanks a lot! Assigning to October so we wrap this up then.

File: docs/nodejs/nodejs-testing.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Testing Node.js
```
Comment (marketicentery) [inline docs/nodejs/nodejs-testing.md:1]: gjhmjgyti8yiulkgjvhdter6tiutiuyuyiut7ytjytuytujyru6uhygkhvmnvjhgvlkup9iy89686

**Review by the1leo — APPROVED** [review]
> This PR adds a skeleton for the Testing doc, so we have a location to make progress on this.

@connor4312 @gregvanl where do you think should this document live? Currently I put it under Typescript, but all of this also holds true for Javascript. We do not have a big Javascript section, only nodejs? So alternatively it could be under nodejs, and the Typescript one would only link to that. What do you think?

@marcusfelling said he can help with the Playwright content

@connor4312 I added sections for the popular testing frameworks, if you think something else should also be covered let me know.

fyi @danielrosenwasser

**Review by Victorbp84 — APPROVED** [review]

**Review by Dtavietnam — APPROVED** [review]

**Review by Dtavietnam — APPROVED** [review]

**Review by Dtavietnam — APPROVED** [review]

**Review by Dtavietnam — APPROVED** [review]
1

**Review by Doni108doni108 — APPROVED** [review]

**Review by rondales — APPROVED** [review]

**Comment by Dtavietnam** [comment]
#6955

**Review by maanu1234 — COMMENTED** [review]
👍

**Review by Abdulhalem-M-hotmail — APPROVED** [review]

**Review by Abdulhalem-M-hotmail — APPROVED** [review]

**Review by Abdulhalem-M-hotmail — APPROVED** [review]

**Review by Abdulhalem-M-hotmail — APPROVED** [review]

**Comment by isidorn** [comment]
@ntrogh championed this effort. Thus closing as done!

**Pull request closed without merging**